src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs

```diff
@@ -18,24 +18,46 @@ public class LogValuesFormatter
         private const string NullValue = "(null)";
         private static readonly object[] EmptyArray = new object[0];
         private static readonly char[] FormatDelimiters = {',', ':'};
-        private readonly string _format;
-        private readonly List<string> _valueNames = new List<string>();
+        private string _format;
+        private List<string> _valueNames = new List<string>();
 
         public LogValuesFormatter(string format)
         {
             OriginalFormat = format;
+        }
 
+        public string OriginalFormat { get; private set; }
+        public List<string> ValueNames
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_format)) FormatInput(OriginalFormat); //construct the valuenames from the input
+                return _valueNames;
+            }
+        }
+
+        private string FormatInput(string format)
+        {
+            if(!string.IsNullOrEmpty(_format))
+            {
+                return _format;
+            }
             var sb = new StringBuilder();
             var scanIndex = 0;
             var endIndex = format.Length;
+            char openBrace = '{', closingBrace = '}';
 
-            while (scanIndex < endIndex)
+            //There should be an open and a closed brace, absence of which, results in unwanted processing.
+            if((!format.Contains(openBrace) && !format.Contains(closingBrace))
```

Comment:
Previous code was optimized to traverse string only ones in search of `{` or `}`, new code would do it multiple times.

In the case where the string doesn't contain braces `_format` would never be set and string would get re-scanned every time `FormatInput` is called.

Reply:
Sorry, the check made only considering the negative case. Since the contains will cause additional search operation, the conditions are removed. 

Also, the _format is set even if the input string does not contain any braces.

Please review the check-in and share your comments.